frontend/ui/impl/src/test/java/com/virtuslab/gitmachete/frontend/ui/impl/GitMacheteErrorReportSubmitterTest.java

```diff
@@ -0,0 +1,42 @@
+package com.virtuslab.gitmachete.frontend.ui.impl;
+
+import static org.mockito.ArgumentMatchers.*;
+
+import java.util.HashMap;
+
+import com.intellij.diagnostic.AbstractMessage;
+import com.intellij.diagnostic.IdeaReportingEvent;
+import com.intellij.diagnostic.LogMessage;
+import com.intellij.openapi.diagnostic.IdeaLoggingEvent;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.reflect.Whitebox;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({GitMacheteErrorReportSubmitter.class})
+public class GitMacheteErrorReportSubmitterTest {
+  @Test
+  public void testGitHubIssueUriConstruction() throws Exception {
+    IdeaLoggingEvent[] events = new IdeaLoggingEvent[1];
+
+    AbstractMessage messageObject = Whitebox.newInstance(LogMessage.class);
+
+    events[0] = new IdeaReportingEvent(messageObject, /* message */ null, /* stacktrace */"", /* plugin */null);
+
+    GitMacheteErrorReportSubmitter reportSubmitter = PowerMockito.spy(new GitMacheteErrorReportSubmitter());
+
+    PowerMockito
+        .stub(PowerMockito.method(GitMacheteErrorReportSubmitter.class, "getTemplateVariables", IdeaLoggingEvent[].class,
+            String.class))
+        .toReturn(new HashMap<String, String>());
+
+    String reportBody = Whitebox.invokeMethod(reportSubmitter, "getReportBody", events, /* additionalInfo */ null);
+
+    Whitebox.invokeMethod(reportSubmitter, "constructNewGitHubIssueUri", events, reportBody);
+
+    Whitebox.invokeMethod(reportSubmitter, "constructNewGitHubIssueUri", events, null);
```

Comment:
Hmmm the result isn't checked against the expected value? 🤔 as for now, it's basically only checked whether a runtime exception happens

Reply:
Well, we could test if it matches the URI we expect, but then we would need to hard-code the expected URI, or create it using same logic as in the tested class. So is there a point?

Reply:
Hardcode the expected URI then, definitely. It can be even be stored in module's `src/test/resources`.

This enitire class is quite important (since it allows users to well, report errors in the plugin smoothly), it's vital that we spot regressions as soon as possible.

Reply:
Also, this'll be be pretty important to provide a regression test for #1008